lib/nsq/connection.ex

```diff
@@ -62,18 +63,15 @@ defmodule NSQ.Connection do
   # ------------------------------------------------------- #
   # Behaviour Implementation                                #
   # ------------------------------------------------------- #
-  @spec start_link(pid, host_with_port, NSQ.Config.t(), String.t(), String.t(), pid, list) ::
+  @spec start_link(pid | host_with_port | NSQ.Config.t() | String.t() | String.t() | pid) ::
```

Comment:
This spec says you receive a single argument that is any one of these. However, you actually receive a list with each of these positionally located in that list. The correct spec would be the following:

 ```suggestion
  @spec start_link([pid, host_with_port, NSQ.Config.t(), String.t(), String.t(), pid]) ::
```

It looks like you have the same problem in a couple other places that you should fix as well.

Reply:
Hey, good one! 
I was scratching my head on this a bit.
If I do this (at a different function but same at all) I get this error when I compile:
```
unexpected list in typespec: [binary(), NSQ.Config.t()]
```

Reply:
I think the simplest here is to use tuples instead of the lists. 
If you know how to make the list work let me know!
Cheers

Reply:
Yea, I think tuple is probably what's appropriate here.